File: Smart_Freight.Server/Models/Product.cs
namespace Smart_Freight.Server.Models;

public class Product
{
    public Guid Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public string? Sku { get; set; }
    public decimal UnitWeightKg { get; set; }
    public DateTimeOffset CreatedAt { get; set; } = DateTimeOffset.UtcNow;

    public ProductStock? Stock { get; set; }
}

File: smart_freight.client/src/types/product.ts
export type Product = {
    id: string;
    name: string;
    sku?: string | null;
    unitWeightKg: number;
    availableQuantity: number;
    createdAt: string;
};

export type ProductCreateRequest = {
    name: string;
    sku?: string | null;
    unitWeightKg: number;
};

export type ProductUpdateRequest = ProductCreateRequest;

export type ProductStock = {
    productId: string;
    availableQuantity: number;
    updatedAt: string;
};

export type StockAdjustRequest = {
    quantityChange: number;
    reason: string;
};
